src/useErrorBoundary.ts

```diff
@@ -2,6 +2,10 @@ import { useContext, useMemo, useState } from "react";
 import { assertErrorBoundaryContext } from "./assertErrorBoundaryContext";
 import { ErrorBoundaryContext } from "./ErrorBoundaryContext";
 
+type UseErrorBoundaryState<Error> =
+  | { error: Error; hasError: true }
+  | { error: null; hasError: false };
+
```

Comment:
If we are aiming for more clarity here, the error type should be “any” for the true case. (Anything- even null- can be thrown in JavaScript.)

Reply:
This `Error` is not JavaScript Error, just **generic name**.
how do you think rename generic `Error` to `TError` like @tanstack/react-query's `TError`, `TData`. I think this renaming can make developer can avoid misunderstading it as JavaScript Error

I pushed a commit to do it in [404f9f9](https://github.com/bvaughn/react-error-boundary/pull/161/commits/404f9f9d02413637fdfdbf5809ddb16be390cecc)

then I guess we can intuitively understand `TError` will be type `any`.

Reply:
I just read the diff on my mobile, and yes, I misread `Error` as a JavaScript `Error`